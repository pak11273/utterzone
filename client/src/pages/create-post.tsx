import { Form, Formik } from "formik"

import { InputField } from "../components/InputField"
import { Layout } from "../components/Layout"
import React from "react"
import { useCreatePostMutation } from "../generated/graphql"
import { useIsAuth } from "../utils/useIsAuth"
import { withApollo } from "../utils/withApollo"

const CreatePost: React.FC<{}> = ({}) => {
  useIsAuth()
  const [createPost] = useCreatePostMutation()
  return (
    <Layout variant="small">
      <Formik
        initialValues={{ title: "", text: "" }}
        onSubmit={async values => {
          const { errors } = await createPost({
            variables: { input: values },
            update: (cache: any) => {
              cache.evict({ fieldName: "posts:{}" })
            },
          })
          if (!errors) {
            // router.push("/")
          }
        }}
      >
        {/* {({ isSubmitting }) => ( */}
        {() => (
          <Form>
            <InputField name="title" placeholder="title" label="Title" />
            <div>
              <InputField
                textarea
                name="text"
                placeholder="text..."
                label="Body"
              />
            </div>
            <button type="submit">create post</button>
          </Form>
        )}
      </Formik>
    </Layout>
  )
}

export default withApollo({ ssr: false })(CreatePost)
