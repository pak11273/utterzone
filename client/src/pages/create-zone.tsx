/* eslint-disable no-template-curly-in-string */
/* eslint-disable @typescript-eslint/no-unused-vars */

import { Button, Checkbox, Form, Input, InputNumber, Select } from "antd"
import React, { useState } from "react"
import { gql, useLazyQuery, useQuery } from "@apollo/client"

import { Language } from "../data/language"
import { cache } from "../apollo/apollo"
import { loader } from "graphql.macro"
import { useCreateZoneMutation } from "../generated/graphql"
import { useHistory } from "react-router-dom"
import { v4 } from "uuid"

const CREATE_ZONE = loader("../graphql/mutations/createZone.graphql")
const Me = loader("../graphql/queries/me.graphql")

// import { toErrorMap } from "../utils/toErrorMap"

const layout = {
  labelCol: { span: 12 },
  wrapperCol: { span: 24 },
}

const validateMessages = {
  required: "${label} is required!",
  types: {
    email: "${label} is not validate email!",
    number: "${label} is not a validate number!",
  },
  number: {
    range: "${label} must be between ${min} and ${max}",
  },
}

const { Option } = Select

interface CreateZoneProps {}

const CACHE_ME = gql`
  query {
    me {
      id
      username
    }
  }
`

const tips = "Max 30 people.  May vary with apps."

export const CreateZone: React.FC<CreateZoneProps> = () => {
  const { data } = useQuery(CACHE_ME)
  const history = useHistory()
  const [privateZone, setPrivate] = useState(false)
  const [getMe] = useLazyQuery(CACHE_ME)
  const [createZoneMutation, { loading }] = useCreateZoneMutation()

  cache.writeQuery({
    query: Me,
    data: {
      isLoggedIn: !!localStorage.getItem("uzid"),
    },
  })

  const onFinish = async (values: any) => {
    if (values) {
      try {
        if (!data.me) {
          console.log('login" ', "you need to log in")
        }

        values.hostname = await data.me.username

        const response = await createZoneMutation({
          variables: { input: values },
        })
        console.log("response: ", response)

        if (response) {
          history.push(
            `/zone/${response.data?.createZone.id}/${response.data?.createZone.token}`
          )
        }
      } catch (err) {
        console.log("catch: ", err)
      }
    }
  }

  const [form] = Form.useForm()

  return (
    <section className="userform_section">
      <div className="user_form">
        <h1>Create a Zone</h1>
        <Form
          validateMessages={validateMessages}
          {...layout}
          layout="vertical"
          form={form}
          name="register"
          onFinish={onFinish}
          scrollToFirstError
          initialValues={{
            name: "A Random Zone",
            maxParticipants: 30,
            nativeLanguage: "Korean",
            learningLanguage: "English",
            app: "lobby",
          }}
        >
          <Form.Item
            name="name"
            label="Name"
            rules={[
              {
                required: true,
                message: "Zones require names",
              },
              {
                min: 8,
                message: "Must be a min of 8 chars",
              },
              {
                max: 24,
                message: "Must be a max of 24 chars",
              },
            ]}
          >
            <Input placeholder="Enter a zone name" />
          </Form.Item>
          <Form.Item
            name="nativeLanguage"
            label="Native Language"
            rules={[{ required: true }]}
          >
            <Select
              allowClear
              showSearch
              style={{ width: 200 }}
              placeholder="Select a language"
              optionFilterProp="children"
              filterOption={(input, option) =>
                option!.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
              }
            >
              <>
                {Language.map((lang, i) => (
                  <Option key={i} value={`${lang}`}>
                    {lang}
                  </Option>
                ))}
              </>
            </Select>
          </Form.Item>
          <Form.Item
            name="learningLanguage"
            label="Learning Language"
            rules={[{ required: true }]}
          >
            <Select
              allowClear
              showSearch
              style={{ width: 200 }}
              placeholder="Select a language"
              optionFilterProp="children"
              filterOption={(input, option) =>
                option!.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
              }
            >
              <>
                {Language.map((lang, i) => (
                  <Option key={i} value={`${lang}`}>
                    {lang}
                  </Option>
                ))}
              </>
            </Select>
          </Form.Item>
          <Form.Item
            name="maxParticipants"
            label="Zone Max"
            help={tips}
            rules={[
              {
                required: true,
                message: "This field is required",
              },
            ]}
          >
            <InputNumber min={1} max={30} />
          </Form.Item>
          <Form.Item name="app" label="app">
            <Select
              allowClear
              showSearch
              style={{ width: 200 }}
              placeholder="Select an app"
              optionFilterProp="children"
              filterOption={(input, option) =>
                option!.children.toLowerCase().indexOf(input.toLowerCase()) >= 0
              }
            >
              <>
                {["youtube", "lobby"].map((app, i) => (
                  <Option key={i} value={`${app}`}>
                    {app}
                  </Option>
                ))}
              </>
            </Select>
          </Form.Item>
          <div style={{ display: "flex", marginTop: "20px" }}>
            <Form.Item
              name="public"
              valuePropName="Private"
              labelAlign="right"
              style={{ marginRight: "40px" }}
            >
              <Checkbox onChange={() => setPrivate(!privateZone)}>
                Private
              </Checkbox>
            </Form.Item>
            <Form.Item name="mature" valuePropName="Mature" labelAlign="right">
              <Checkbox>18+</Checkbox>
            </Form.Item>
          </div>
          {privateZone ? (
            <>
              <Form.Item
                label="Password"
                name="password"
                rules={[
                  {
                    required: true,
                    message: "Passwords required for private zones",
                  },
                ]}
              >
                <Input.Password />
              </Form.Item>
              <Form.Item
                name="confirm"
                label="Confirm Password"
                dependencies={["password"]}
                hasFeedback
                rules={[
                  {
                    required: true,
                    message: "Please confirm your password!",
                  },
                  ({ getFieldValue }) => ({
                    validator(rule, value) {
                      if (!value || getFieldValue("password") === value) {
                        return Promise.resolve()
                      }
                      return Promise.reject(
                        "The two passwords that you entered do not match!"
                      )
                    },
                  }),
                ]}
              >
                <Input.Password />
              </Form.Item>
            </>
          ) : null}
          <Form.Item>
            <Button
              style={{ margin: "20px 0 0 6px" }}
              type="primary"
              htmlType="submit"
              loading={loading}
              onClick={() => getMe()}
            >
              Create Zone
            </Button>
          </Form.Item>
        </Form>
      </div>
    </section>
  )
}
