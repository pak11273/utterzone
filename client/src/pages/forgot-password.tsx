import { Form, Formik } from "formik"
import React, { useState } from "react"

import { InputField } from "../components/InputField"
import { useForgotPasswordMutation } from "../generated/graphql"
import { withApollo } from "../utils/withApollo"

const ForgotPassword: React.FC<{}> = ({}) => {
  const [complete, setComplete] = useState(false)
  const [forgotPassword] = useForgotPasswordMutation()
  return (
    <Formik
      initialValues={{ email: "" }}
      onSubmit={async values => {
        await forgotPassword({ variables: values })
        setComplete(true)
      }}
    >
      {/* {({ isSubmitting }) => */}
      {() =>
        complete ? (
          <div>if an account with that email exists, we sent you can email</div>
        ) : (
          <Form>
            <InputField
              name="email"
              placeholder="email"
              label="Email"
              type="email"
            />
            <button type="submit">forgot password</button>
          </Form>
        )
      }
    </Formik>
  )
}

export default withApollo({ ssr: false })(ForgotPassword)
