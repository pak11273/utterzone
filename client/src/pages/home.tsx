import { AdHeader } from "../components"
import { Link } from "react-router-dom"
import React from "react"

export const Home = (props: any) => {
  return (
    <>
      <AdHeader />
      <div className="home">
        <div className="home_container">
          <h1>Welcome to Utterzone</h1>
          <h2>Learn to speak a new language</h2>
          <div className="home_guides">
            <h2>Guides</h2>
            <section>
              <div>
                <Link to="#">Utterzone Tutorials</Link>
              </div>
              <div>
                <Link to="#">Chat Etiquette</Link>
              </div>
              <div>
                <Link to="#">Learning Efficiently</Link>
              </div>
            </section>
          </div>
        </div>
      </div>
    </>
  )
}
