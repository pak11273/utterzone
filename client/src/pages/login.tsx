import { Button, Form, Input } from "antd"
import { Link, useHistory } from "react-router-dom"
import { MeDocument, MeQuery, useLoginMutation } from "../generated/graphql"

import React from "react"
import { toErrorMap } from "../utils/toErrorMap"
import { withRouter } from "react-router-dom"

// import { SwitchField } from "../components"

// const { Option } = Select
const formItemLayout = {}

const Container = (props: any) => {
  const [login] = useLoginMutation()
  const history = useHistory()
  const [form] = Form.useForm()

  const onFinish = async (values: any) => {
    if (values) {
      const response = await login({
        variables: values,
        update: (cache, { data }) => {
          cache.writeQuery<MeQuery>({
            query: MeDocument,
            data: {
              __typename: "Query",
              me: data?.login.user,
            },
          })
          cache.evict({ fieldName: "posts:{}" })
        },
      })
      if (response.data?.login.errors) {
        const errorMap = toErrorMap(response.data?.login.errors)
        form.setFields(errorMap)
      }
      if (response.data?.login.user) {
        history.push("/")
      }
    }
  }

  const onFinishFailed = ({
    values,
    errorFields,
    outOfDate,
  }: {
    values: any
    errorFields: any
    outOfDate: any
  }) => {
    console.log("val: ", values)
    console.log("errors: ", errorFields)
    console.log("out: ", outOfDate)
  }

  return (
    <section className="userform_section">
      <div className="user_form">
        <Form
          {...formItemLayout}
          layout="vertical"
          form={form}
          name="register"
          onFinish={onFinish}
          onFinishFailed={onFinishFailed}
          initialValues={{
            residence: ["zhejiang", "hangzhou", "xihu"],
            prefix: "86",
          }}
          scrollToFirstError
        >
          <Form.Item
            name="usernameOrEmail"
            label="E-mail"
            rules={[
              {
                type: "email",
                message: "The input is not valid E-mail!",
              },
              {
                required: true,
                message: "Please input your E-mail!",
              },
            ]}
          >
            <Input />
          </Form.Item>

          <Form.Item
            label="Password"
            name="password"
            rules={[{ required: true, message: "Please input your password!" }]}
          >
            <Input.Password />
          </Form.Item>
          <Form.Item>
            <Button
              style={{ margin: "20px 0 0 6px" }}
              type="primary"
              htmlType="submit"
            >
              Log In
            </Button>
            <Button type="link" htmlType="button">
              <Link to="/forgot-password">forgot password?</Link>
            </Button>
          </Form.Item>
        </Form>
      </div>
    </section>
  )
}

export const Login = withRouter(Container)
