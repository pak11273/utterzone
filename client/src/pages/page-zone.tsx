import { Chat, Zone, ZoneControls } from "../components"
import { Col, Row } from "antd"

import React from "react"

interface zoneProps {}

export const PageZone: React.FC<zoneProps> = () => {
  return (
    <div style={{ height: "100%" }}>
      <Row style={{ height: "100%" }}>
        <Col
          className="zone-content"
          span={17}
          style={{ background: "green", height: "100%", overflow: "scroll" }}
        >
          <Zone />
          <ZoneControls />
        </Col>
        <Col span={7} style={{ height: "100%" }}>
          <Chat />
        </Col>
      </Row>
    </div>
  )
}
