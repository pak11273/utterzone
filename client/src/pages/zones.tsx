// import { MeDocument, MeQuery } from "../generated/graphql"

// import { Form } from "antd"

import { Col, Row } from "antd"
import React, { useEffect } from "react"

import { Button } from "antd"
import { ZoneCard } from "../components"
import { useZonesQuery } from "../generated/graphql"

// import { useHistory } from "react-router-dom"

// import { register } from "../serviceWorker"
// import { toErrorMap } from "../utils/toErrorMap"

interface zonesProps {}

// const { Meta } = Card

export const Zones: React.FC<zonesProps> = () => {
  // const history = useHistory()
  // const [form] = Form.useForm()
  const { data, loading } = useZonesQuery({
    variables: {},
    fetchPolicy: "no-cache",
  })

  // if (loading) {
  //   setLoading(true)
  // }

  useEffect(data => {
    if (!data || !data.zones) {
      console.log("no zones loaded!")
    }
  }, [])

  // const onFinish = async (values: any) => {
  // if (values) {
  //   const response = await register({
  //     variables: { options: values },
  //     update: (cache, { data }) => {
  //       cache.writeQuery<MeQuery>({
  //         query: MeDocument,
  //         data: {
  //           __typename: "Query",
  //           me: data?.register.user,
  //         },
  //       })
  //     },
  //   })
  //   if (response.data?.register.errors) {
  //     const errorMap = toErrorMap(response.data?.register.errors)
  //     form.setFields(errorMap)
  //   } else {
  //     history.push("/")
  //   }
  // }
  // }
  console.log("data: ", data)

  return (
    <section className="userform_section">
      <div className="zones_container">
        <div
          style={{
            display: "flex",
            flexDirection: "column",
            justifyContent: "center",
          }}
        >
          <h1 style={{ textAlign: "center" }}>Filters</h1>
          <Row gutter={[24, 32]}>
            <Col xs={24}>I speak: English</Col>
            <Col xs={24}>Learning: Korean</Col>
            <Col xs={24}>Participants: High to Low</Col>
            <Col xs={24}>App: Youtube</Col>
            {/* <Col xs={24}>Mature Zone</Col> */}
          </Row>
          <Button size="small" type="primary" style={{ width: "100px" }}>
            Select
          </Button>
        </div>
        <Row gutter={[24, 32]}>
          {data
            ? data?.zones.map((x, i) => {
                return (
                  <Col key={i} xs={24} sm={18} md={12} lg={6} xl={4}>
                    <ZoneCard
                      app={x.app}
                      name={x.name}
                      loading={loading}
                      description={x.description}
                      hostId={x.hostId}
                      zoneId={x.zoneId}
                      premium={x.premium}
                      mature={x.mature}
                    />
                  </Col>
                )
              })
            : null}
        </Row>
      </div>
    </section>
  )
}
