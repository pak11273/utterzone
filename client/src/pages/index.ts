import { ForgotPassword } from "./forgot-password"
import { Home } from "./home"
import { Login } from "./login"
import { Register } from "./register"
import { Zone } from "./zone"
import { ZoneCreate } from "./zone-create"
import { Zones } from "./zones"

export { ForgotPassword, Home, Login, Register, Zone, Zones, ZoneCreate }
