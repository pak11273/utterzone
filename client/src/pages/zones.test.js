import { BrowserRouter } from "react-router-dom"
import { MockedProvider } from "@apollo/client/testing"
import React from "react"
import TestRenderer from "react-test-renderer"
import { Zones } from "../pages/zones"
import { loader } from "graphql.macro"
import puppeteer from "puppeteer"

const zones = loader("../graphql/queries/zones.graphql")
// const login = loader("../graphql/mutations/login.graphql")
// fixes the window.matchMedia is not a function error
beforeAll(() => {
  Object.defineProperty(window, "matchMedia", {
    value: jest.fn(() => {
      return {
        matches: true,
        addListener: jest.fn(),
        removeListener: jest.fn(),
      }
    }),
  })
})

const mocks = [
  {
    request: {
      query: zones,
    },
    result: {
      data: {
        zones: { id: "1", name: "Buck", breed: "bulldog" },
      },
    },
  },
]

describe("/zones/all", () => {
  const ZonesComponent = (
    <MockedProvider mocks={mocks} addTypename={false}>
      <BrowserRouter>
        <Zones />
      </BrowserRouter>
    </MockedProvider>
  )
  const comp = TestRenderer.create(ZonesComponent)

  it("render without crashing and match its snapshot", () => {
    let tree = comp.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should display text that no zones were found if response is empty", () => {})
})
