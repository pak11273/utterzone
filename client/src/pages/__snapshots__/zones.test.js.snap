// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/zones/all render without crashing and match its snapshot 1`] = `
<section
  className="userform_section"
>
  <div
    className="zones_container"
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": "60px",
        }
      }
    >
      <h1
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Criteria
      </h1>
      <div
        className="ant-row"
        style={
          Object {
            "marginBottom": 16,
            "marginLeft": -12,
            "marginRight": -12,
            "marginTop": -16,
          }
        }
      >
        <div
          className="ant-col ant-col-xs-24 ant-col-md-4"
          style={
            Object {
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          I speak: English
        </div>
        <div
          className="ant-col ant-col-xs-24 ant-col-md-4"
          style={
            Object {
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          Learning: Korean
        </div>
        <div
          className="ant-col ant-col-xs-24 ant-col-md-4"
          style={
            Object {
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          Participants: High to Low
        </div>
        <div
          className="ant-col ant-col-xs-24 ant-col-md-4"
          style={
            Object {
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          App: Youtube
        </div>
        <div
          className="ant-col ant-col-xs-24 ant-col-md-4"
          style={
            Object {
              "paddingBottom": 16,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 16,
            }
          }
        >
          Mature Zone
        </div>
      </div>
      <button
        className="ant-btn ant-btn-primary ant-btn-sm"
        onClick={[Function]}
        style={
          Object {
            "width": "100px",
          }
        }
        type="button"
      >
        <span>
          Select
        </span>
      </button>
    </div>
    <div
      className="ant-row"
      style={
        Object {
          "marginBottom": 16,
          "marginLeft": -12,
          "marginRight": -12,
          "marginTop": -16,
        }
      }
    />
  </div>
</section>
`;
